x-pack/elastic-agent/pkg/core/plugin/process/app.go

```diff
@@ -250,3 +274,31 @@ func (a *Application) setState(s state.Status, msg string, payload map[string]in
 func (a *Application) cleanUp() {
 	a.monitor.Cleanup(a.desc.Spec(), a.pipelineID)
 }
+
+func gracefulKill(proc *process.Info) {
+	if proc == nil || proc.Process == nil {
+		return
+	}
+
+	// send stop signal to request stop
+	proc.Process.Signal(os.Interrupt)
```

Comment:
Does this work on Windows? I thought `process.Info` had each OS implementation that is why it was added. Should we just move this logic into the `Stop()` function of `process.Info`?

Reply:
you're right windows does not implement sending interupt. using Stop function instead